### ISSUE TITLE
Write API from our supernet to LM-Eval-Harness

### ISSUE THREAD

**Issue by rheasukthanker** [whittle-org/whittle#46]
LiTGPT writes checkpoints to a directory, converts it into an HF model and evaluates it. We evaluate a lot of models during search and want this to be as seamless as possible. Writing an api for our models in lm-eval-harness would make evaluation during search much faster. Following are the list of checks necessary to integrate this:
1. Write a custom API following https://github.com/EleutherAI/lm-evaluation-harness/blob/main/docs/interface.md#external-library-usage
2. Test that predictions/logits for an HF model and our model matches with a pretrained checkpoint
